core/src/local/persistent_state.rs

```diff
@@ -279,6 +297,9 @@ impl PersistentState {
         });
 
         let full_name = format!("{}.db", name);
+        // We use a column for each index, and one for meta information.
+        // When opening the DB the exact same column families needs to be used,
+        // so we'll have to retrieve the existing ones first:
         let column_family_names = match DB::list_cf(&opts, &full_name) {
             Ok(cfs) => cfs,
```

Comment:
Couldn't we do the thing you do in `add_key` here if `cfs.len()` != meta.indices.len()`?